### DISCUSSION TITLE
using process.env.XXX leads to config validation warning

### DISCUSSION THREAD

**Discussion by pehbehbeh** [renovatebot/renovate#29470] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

We are using the following global config:

```js
module.exports = {
    "secrets": {
        RENOVATE_TOKEN: process.env.RENOVATE_TOKEN
    },
    // ...
    "hostRules": [
        {
            "hostType": "docker",
            "matchHost": "registry.company.com",
            "username": "renovate",
            "password": "{{ secrets.RENOVATE_TOKEN }}",
        }
    ]
}
```

But the config validator outputs an warning:

```
$ renovate-config-validator $RENOVATE_CONFIG_VALIDATOR_EXTRA_FLAGS
 INFO: Validating .gitlab/renovate.json
 WARN: Config validation warnings found in config.js
       "warnings": [
         {
           "topic": "Configuration Error",
           "message": "Invalid `secrets.RENOVATE_TOKEN` configuration: value must be a string."
         }
       ]
 INFO: Validating config.js
 WARN: Found errors in configuration
       "file": "config.js",
       "warnings": [
         {
           "topic": "Configuration Error",
           "message": "Invalid `secrets.RENOVATE_TOKEN` configuration: value must be a string."
         }
       ]
```

We get the same error when directly using `"password": process.env.RENOVATE_TOKEN`.

**Comment by pehbehbeh** [comment]
It is also quite strange that this warning only exists in Renovate MR branches. As soon as I merge these, the error no longer appears in the main branch. The `.gitlab-ci.yml` is as simple as:

```yml
variables:
  RENOVATE_EXTRA_FLAGS: --autodiscover=true --onboarding=false

include:
  - remote: https://gitlab.com/renovate-bot/renovate-runner/-/raw/v17.241.1/templates/renovate.gitlab-ci.yml
  - remote: https://gitlab.com/renovate-bot/renovate-runner/-/raw/v17.241.1/templates/renovate-config-validator.gitlab-ci.yml
```

And the `renovate.json` looks like that:

```json
{
    "extends": [
        "gitlab>renovate-bot/renovate-runner"
    ],
    "reviewers": [
        "..."
    ]
}
```

Logs from the main branch:

```
$ renovate-config-validator $RENOVATE_CONFIG_VALIDATOR_EXTRA_FLAGS
 INFO: Validating .gitlab/renovate.json
 INFO: Validating config.js
 INFO: Config validated successfully
```

**Reply by pehbehbeh** [reply]
Just solved it myself: `RENOVATE_TOKEN` was a protected CI variable (only available in the main branch). But maybe that should not lead to a validation error? 🤔 After all, it is valid syntax.

**Reply by rarkins** [reply, marked as answer]
It's intentional that if someone has a rule pointing to an undefined env variable, and asks us to validate, then we warn them
